pkg/cmd/util/clientcmd/factory.go

```diff
@@ -192,9 +192,14 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 				return nil, errors.New("provided options object is not a BuildLogOptions")
 			}
 			if bopts.Version != nil {
-				// should --version work with builds at all?
 				return nil, errors.New("cannot specify a version and a build")
 			}
+			if bopts.Previous {
+				version := buildutil.VersionForBuild(t.Name)
+				version--
+				name := buildutil.ConfigNameForBuild(t.Name)
```

Comment:
you should get the configname from the annotation on the build object instead of by parsing.